lib/puppet/transaction.rb

```diff
@@ -302,6 +310,13 @@ def failed_dependencies?(resource)
   # up-front at failure time because the graph may be mutated as we
   # walk it.
   def propagate_failure(resource)
+
+    provider_class = resource.provider.class
+    if(prefetch_failed_providers[resource.type][provider_class.name] and s = resource_status(resource))
+      message = _("Prefetch failed earlier for %{type_name} provider '%{name}'") % { type_name: resource.type, name: provider_class.name }
```

Comment:
From the users perspective reading this error message does "earlier" matter?

Reply:
"earlier" should probably be dropped. It means "at some point in the past during this catalog run", but that's not adding a ton of useful information. We can say "`Prefetch failed for '%{type_name}' provider '%{provider_name}'`" and that should be sufficient.